### PR TITLE
ccxt update, Bybitv3 and handle canceling order state

### PR DESCRIPTION
**Pull request by GuillaumeDSM**

### CODE REVIEW

File: octobot_trading/personal_data/orders/states/cancel_order_state.py [hunk 1]
```diff
@@ -48,12 +50,21 @@ def is_status_pending(self) -> bool:
     def is_status_cancelled(self) -> bool:
         return not self.is_status_pending() and self.order.status in constants.CANCEL_ORDER_STATUS_SCOPE
 
+    def allows_new_status(self, status) -> bool:
```
Comment (Herklos) [inline octobot_trading/personal_data/orders/states/cancel_order_state.py:53]: :+1:

File: octobot_trading/personal_data/orders/types/market/sell_market_order.py [hunk 2]
```diff
@@ -18,5 +18,5 @@
 
 
 class SellMarketOrder(market_order.MarketOrder):
-    def __init__(self, trader):
+    def __init__(self, trader, side=None):
```
Comment (Herklos) [inline octobot_trading/personal_data/orders/types/market/sell_market_order.py:21]: ?

File: octobot_trading/personal_data/positions/channel/positions_updater.py [hunk 3]
```diff
@@ -136,7 +137,7 @@ async def fetch_position_per_symbol(self):
                 positions += fetched_positions
 
         if positions:
-            self._initialize_contract_if_necessary(positions)
+            exchange_data.update_contracts_from_positions(self.channel.exchange_manager, positions)
```
Comment (Herklos) [inline octobot_trading/personal_data/positions/channel/positions_updater.py:140]: :+1:

File: octobot_trading/personal_data/state.py [hunk 4]
```diff
@@ -35,6 +37,11 @@ def __init__(self, is_from_exchange_data):
         # state lock
         self.lock = asyncio.Lock()
 
+        # set after self.terminate has been executed (with or without raised exception)
+        self.terminated = asyncio.Event()
+        # seconds between each refresh attempt while in a pending state
+        self.pending_refresh_interval = self.DEFAULT_PENDING_REFRESH_INTERVAL
```
Comment (Herklos) [inline octobot_trading/personal_data/state.py:43]: Should it be modified? If not we can just use a class attribute

File: octobot_trading/personal_data/orders/types/market/sell_market_order.py (hunk 2 above)
Comment (GuillaumeDSM) [reply octobot_trading/personal_data/orders/types/market/sell_market_order.py:21]: buy/sellmarket orders are not respecting the parent class constructor. It creates a need to a special case every time to check if the side should be provided or not, I think it's better to allow the parameter and not use it

File: octobot_trading/enums.py [hunk 5]
```diff
@@ -53,6 +53,7 @@ class EvaluatorStates(enum.Enum):
 class OrderStatus(enum.Enum):
     PENDING_CREATION = "pending_creation"
     OPEN = "open"
+    OPENING = "opening"
```
[the remark this answers is not in the document]
Comment (GuillaumeDSM) [reply octobot_trading/enums.py:56]: It's a mistake, i'm removing it

File: octobot_trading/personal_data/state.py [hunk 6]
```diff
@@ -176,6 +179,13 @@ def on_terminate(self) -> None:
         self.get_logger().debug(f"{self.__class__.__name__} terminated")
         self.terminated.set()
 
+    def __del__(self):
+        if not self.terminated.is_set() and self.terminated._waiters:
```
Comment (Herklos) [inline octobot_trading/personal_data/state.py:183]: :+1:

File: octobot_trading/personal_data/orders/orders_manager.py [hunk 7]
```diff
@@ -51,6 +51,9 @@ def get_all_orders(self, symbol=None, since=-1, limit=-1, tag=None):
     def get_open_orders(self, symbol=None, since=-1, limit=-1, tag=None):
         return self._select_orders(enums.OrderStatus.OPEN, symbol, since, limit, tag=tag)
 
+    def get_pending_cancel_orders(self, symbol=None, since=-1, limit=-1, tag=None):
```
Comment (Herklos) [inline octobot_trading/personal_data/orders/orders_manager.py:54]: :+1:

**Review by Herklos — APPROVED** [review]
:+1:

**Comment by Herklos** [comment]
Great fixes!

**Review by Herklos — APPROVED** [review]

**Pull request merged**